File: plugin/src/main/groovy/com/shogo82148/ribbonizer/GrayScaleBuilder.java
package com.shogo82148.ribbonizer;

import com.android.build.gradle.api.ApplicationVariant;
import com.shogo82148.ribbonizer.filter.GrayScaleFilter;
import com.shogo82148.ribbonizer.resource.Resource;

import java.io.File;
import java.util.function.Consumer;

public class GrayScaleBuilder implements FilterBuilder {

    @Override
    public Consumer<Resource> apply(ApplicationVariant applicationVariant, File iconFile) {
        return new GrayScaleFilter();
    }
}

File: ribbonizer-plugin/src/main/kotlin/com/shogo82148/ribbonizer/GrayScaleBuilder.kt
package com.shogo82148.ribbonizer

import com.android.build.gradle.api.ApplicationVariant
import com.shogo82148.ribbonizer.filter.GrayScaleFilter
import com.shogo82148.ribbonizer.resource.Resource
import com.shogo82148.ribbonizer.resource.Variant
import java.io.File
import java.util.function.Consumer

class GrayScaleBuilder : FilterBuilder {
    override fun apply(variant: Variant, iconFile: File): Consumer<Resource> {
        return GrayScaleFilter()
    }
}